gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -191,7 +193,20 @@ private void showAddFeatureDialog(Pair<ImmutableList<Layer>, Point> args) {
               viewModel.addFeature(layer, point);
               break;
             case POLYGON:
-              showPolygonInfoDialog(layer);
+              if (localValueStore.isPolygonDialogInfoShown()) {
```

Comment:
Please refactor this block and the identical block below into a common method.

Reply:
updated.